src/responsive-layout.tsx

```diff
@@ -41,7 +41,12 @@ export const InternalResponsiveLayout: React.FC<Props & {shouldExpandWhenNested?
                     internalVariant &&
                         internalVariant !== 'default' &&
                         styles.backgroundVariant[internalVariant],
-                    shouldExpandWhenNested && !fullWidth && styles.expandedResponsiveLayoutContainer
+                    {
+                        [styles.expandedResponsiveLayoutContainerMobile]:
+                            shouldExpandWhenNested === true && !fullWidth,
```

Comment:
```suggestion
                            shouldExpandWhenNested && !fullWidth,
```

Reply:
`shouldExpandWhenNested` could be `"desktop"`, which is truthy. 

Perhaps a comment to communicate the intention would be useful

Reply:
remember what I told you the other day @marcoskolodny? nobody writes `=== false` or `=== true` without a reason :D

Reply:
Oooh, I missed the fact that it can be a string. My bad!